### PR TITLE
docs: Add migration guide from version 2.x to 3.x

### LINKED ISSUE #960: v3: add migration helpers and guide for breaking changes
I'd like to make sure we don't get folks stuck on v2. I know we have some internal (Google) customers who rely on the BQML stats protobuf messages (the ones in `google.cloud.bigquery_v2`.

I propose

1. We restore the files in `google.cloud.bigquery_v2`, but document that they are **deprecated** and are **not updated**.
2. We continue to return dictionaries in the relevant classes, so that all stays the same.
3. We document in the `UPGRADING` guide how to manually convert from the dictionaries to the classes in `google.cloud.bigquery_v2` with some code samples.

Regarding the pandas connector changes, we should also document those in the `UPGRADING` guide, ideally with code samples on how to get / convert to the old dtypes if desired.

### PR DESCRIPTION
**Pull request by plamut**
Closes #960.

This is a work-in-progress migration guide. I suggest merging it only after all `v3` features and changes have been merged.

This PR can be regularly updated when new changes are merged into the `v3` branch.

**PR checklist:**
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -13,7 +13,126 @@ limitations under the License.
...
-TODO
+## New Required Dependencies
+
+Some of the previously optional dependencies are now *required* in `3.x` versions of the
+library, namely
+[google-cloud-bigquery-storage](https://pypi.org/project/google-cloud-bigquery-storage/)
+(minimum version `2.0.0`) and [pyarrow](https://pypi.org/project/pyarrow/) (minimum
+version `3.0.0`).
+
+The behavior of some of the package "extras" has thus also changed:
```
Comment (tswast) [inline UPGRADING.md:24]: Let's add a bullet that the `pandas` extra now requires the `db-dtypes` package.

Comment (plamut) [reply UPGRADING.md:24]: Will do.

BTW, I noticed that `db-types` is currently marked as Beta, which implies that that `pandas` extra is also just Beta, technically at least. Should we mention that as well or are we expected to promote `db-types` to stable before we release `v3`?

Comment (tswast) [reply UPGRADING.md:24]: This has been resolved. db-dtypes 1.0.0 has been released https://github.com/googleapis/python-db-dtypes-pandas/pull/94

**Review by tswast — APPROVED** [review]

**Comment by tswast** [comment]
Prerelease deps test failure will be resolved by https://github.com/googleapis/python-bigquery/pull/1175/files#diff-148cef988889bf1536c73a57b8aade66e5fa4841e1878631f605a074a3750dfe

**Pull request merged**